### PR TITLE
feat: check if user is in a session and render component

### PR DESCRIPTION
**Pull request by raynerljm**
- [x] API gateway auth module checks if user is in a room, if yes, include the roomId in the returned object
- [x] if FE receives roomId from the `/auth` endpoint, then render the `Session in progress` component
- [x] Clicking session in progress component returns user to the room page

image

### CODE REVIEW

File: backend/api-gateway/app/libs/middleware.ts [hunk 1]
```diff
@@ -44,14 +45,54 @@ export const authMiddleware: RequestHandler = async (req, res, next) => {
...
-  } else {
-    return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).send(
+    const getUserResponse = await fetch(
+      `${process.env.USERS_SERVICE_URL}/api/users/email/${primaryEmail}`
+    );
+    const getUserData = await getUserResponse.json();
+    const safeParsedUserServiceData = getUserSchema.safeParse(getUserData);
+
+    if (!safeParsedUserServiceData.success) {
+      return res.status(500).send(
```
Comment (Josephaedan) [inline backend/api-gateway/app/libs/middleware.ts:58]: nit: Can use HTTP status code enums instead

File: backend/api-gateway/app/libs/middleware.ts [hunk 2]
```diff
@@ -44,14 +45,54 @@ export const authMiddleware: RequestHandler = async (req, res, next) => {
...
+      next();
+    } else {
+      return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).send(
+        failApiResponse({
+          message: "User is not authorized to perform this action",
+        })
+      );
+    }
+  } catch (error) {
+    return res.status(500).send(
```
Comment (Josephaedan) [inline backend/api-gateway/app/libs/middleware.ts:91]: Same as above

File: backend/api-gateway/app/routes/auth.router.ts [hunk 3]
```diff
@@ -60,9 +61,34 @@ authRouter.get("/", authMiddleware, async (req: Request, res: Response) => {
...
+    } catch (error) {
+      console.error(
+        "Error getting room ID:",
+        error instanceof Error ? error.message : JSON.stringify(error)
+      );
+    }
+
+    return res.send(successApiResponse({ user }));
   } catch (error) {
     return res.status(500).send(
```
Comment (Josephaedan) [inline backend/api-gateway/app/routes/auth.router.ts]: Same as above

**Review by Josephaedan — APPROVED** [review]
LGTM

**Review by joelwongjy — APPROVED** [review]

File: backend/api-gateway/app/routes/auth.router.ts (hunk 3 above)
Comment (raynerljm) [reply backend/api-gateway/app/routes/auth.router.ts]: resolved in [f590ded](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/30/commits/f590dedbebef2f7f74e1a2f29e9e015b9be9b6a8)

File: backend/api-gateway/app/libs/middleware.ts (hunk 2 above)
Comment (raynerljm) [reply backend/api-gateway/app/libs/middleware.ts:91]: resolved in [f590ded](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/30/commits/f590dedbebef2f7f74e1a2f29e9e015b9be9b6a8)

File: backend/api-gateway/app/libs/middleware.ts (hunk 1 above)
Comment (raynerljm) [reply backend/api-gateway/app/libs/middleware.ts:58]: resolved in [f590ded](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/pull/30/commits/f590dedbebef2f7f74e1a2f29e9e015b9be9b6a8)

**Pull request merged**